Refactored rendered rolls for Astral and small fixes around.

adriangaro opened the pull request:
This PR should be targeted to fix #538 and some other issues/improvements I wanted to fix/implement.

This would also fix the issue reported in the astral chat which caused spell cards to not be able to roll damage when using rendered rolls.

adriangaro commented:
So an update with the latest changes in the PR, improved error handling here and there.

Resolved an issue notified by @Alkalithe in discord. His issue stemmed  from the possibility of having multiple entries in the firebase indexedb which didn't store the user auth token.

Changed rendered rolls to set the roll result similar to the way @Redmega recommended in #538.

Rendered rolls now should add the required roll info to the damage button, as the damage button would not be working after refresh/would redirect the user to an non-existent url. This is achieved by trimming the data stored and compressing it in the anchor href. The compression is mandatory as we are still bound by the 2048 character limit.

Some improvement on acquiring the iframe content document. Sometimes the initial reference to the iframe was refreshed and caused some error down the line.

kakaroto commented on src/astral/page-script.js:
```diff
@@ -93,21 +119,32 @@ async function handleRenderedRoll(request) {
+        message = `${message}
+
+
+${parseDescription(request, request.description)}`
```
Indentation issue here

kakaroto commented:
Thanks for the changes. I had a very quick look, not much for me to comment on, you're the maintainer for Astral and it generally looked good enough, so I'll merge it, thanks! 👍 
I'd appreciate if you could list here as a response comment the list of features and bugfixes that went into this PR in the same sort of format I usually do in the changelog, so that I can copy it into the changelog on release date.

Thanks!

adriangaro commented:
Change log for Astral:

- **Bugfix**: Fixed digital dice rolls / rendered rolls for Astral. Reverted to normal behaviour when no dice are rolled.
- **Bugfix**: Fixed Astral access token acquisition in some specific edge cases.
- **Bugfix**: Fixed some issues related to slow loading of chat widget in Astral.
- **Bugfix**: Changed rendered roll formulas in Astral using the syntax recommended by @Redmega.
- **Bugfix**: Fixed some errors related to speak as character functionality.
- **Feature/Bugfix**: Changed implementation of roll buttons (when using digital dice) to work after reloading the page. They should work consistently now.
